store/loader.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/rs/zerolog/log"
+
+	"github.com/Layr-Labs/eigenda/common/geth"
+	eigenda_eth "github.com/Layr-Labs/eigenda/core/eth"
+	"github.com/Layr-Labs/eigenda/encoding/kzg"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	geth_common "github.com/ethereum/go-ethereum/common"
+)
+
+// StoreLoader centralizes dependency initialization
+// It ensures proper typing and avoids redundant logic scattered across functions.
+type StoreLoader struct {
+	ctx     context.Context
+	log     logging.Logger
+	metrics metrics.Metricer
+
+	memConfig *memstore.Config
+
+	// configs
+	managerCfg     Config
+	v1VerifierCfg  verify.Config
+	v1EdaClientCfg clients.EigenDAClientConfig
+	v2ClientCfg    common.V2ClientConfig
+}
+
+func NewStoreLoader(ctx context.Context, cfg Config,
+	v1VerifierCfg verify.Config,
+	v1EdaClientCfg clients.EigenDAClientConfig,
+	v2ClientCfg common.V2ClientConfig,
+	memConfig *memstore.Config,
+	log logging.Logger, metrics metrics.Metricer) *StoreLoader {
+	return &StoreLoader{ctx, log, metrics, memConfig, cfg, v1VerifierCfg, v1EdaClientCfg, v2ClientCfg}
+}
+
+func (d *StoreLoader) loadSecondaries(targets []string, s3Store common.PrecomputedKeyStore, redisStore *redis.Store) []common.PrecomputedKeyStore {
+	stores := make([]common.PrecomputedKeyStore, len(targets))
+
+	for i, target := range targets {
+		switch common.StringToBackendType(target) {
+		case common.RedisBackendType:
+			if redisStore == nil {
+				panic(fmt.Sprintf("Redis backend not configured: %s", target))
+			}
+			stores[i] = redisStore
+		case common.S3BackendType:
+			if s3Store == nil {
+				panic(fmt.Sprintf("S3 backend not configured: %s", target))
+			}
+			stores[i] = s3Store
+		default:
+			panic(fmt.Sprintf("Invalid backend target: %s", target))
+		}
+	}
+	return stores
+}
+
+func (d *StoreLoader) loadEigenDAV2Store() (*eigendav2.Store, error) {
+	gethCfg := geth.EthClientConfig{
+		RPCURLs: []string{d.v1EdaClientCfg.EthRpcUrl},
+	}
+
+	ethClient, err := geth.NewClient(gethCfg, geth_common.Address{}, 0, d.log)
+	if err != nil {
+		return nil, err
+	}
+
+	g1Points, err := kzg.ReadG1Points(
+		d.v1VerifierCfg.KzgConfig.G1Path,
+		d.v1VerifierCfg.KzgConfig.SRSNumberToLoad, 4)
+	if err != nil {
+		return nil, err
+	}
+
+	reader, err := eigenda_eth.NewReader(d.log, ethClient, "0x0", d.v2ClientCfg.ServiceManagerAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	relayURLs, err := reader.GetRelayURLs(d.ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	relayCfg := clients_v2.RelayClientConfig{
+		UseSecureGrpcFlag:  true,
+		MaxGRPCMessageSize: d.v2ClientCfg.PutRetries,
+		Sockets:            relayURLs,
+	}
+
+	retriever, err := clients_v2.BuildRelayPayloadRetriever(d.log, d.v2ClientCfg.RetrievalConfig, &relayCfg, g1Points)
+	if err != nil {
+		return nil, err
+	}
+
+	disperser, err := clients_v2.BuildPayloadDisperser(d.log, d.v2ClientCfg.PayloadClientCfg, &d.v2ClientCfg.DisperserClientCfg, &gethCfg, nil, nil)
```

Comment:
@litt3 I thought we had embedded the disperser config into the payloadDisperserConfig?
Doesn't it feel a bit weird to have to pass 2 configs here? Feels like the abstraction is leaking